iothub/service/src/Configurations/ConfigurationsClient.cs

```diff
@@ -241,7 +241,7 @@ public virtual async Task<Configuration> SetAsync(Configuration configuration, b
                 cancellationToken.ThrowIfCancellationRequested();
 
                 using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Put, GetConfigurationRequestUri(configuration.Id), _credentialProvider, configuration);
-                HttpMessageHelper.InsertETag(request, configuration.ETag);
+                HttpMessageHelper.ConditionallyInsertETag(request, configuration.ETag);
```

Comment:
You aren't passing along the `forceUpdate` parameter.

Reply:
This PR is just for DevicesClient. We'll fix this later